src/quick-lint-js/parse.h

```diff
@@ -2443,6 +2443,8 @@ class parser {
     QLJS_ASSERT(this->peek().type == token_type::kw_with);
     this->skip();
 
+    v.visit_enter_with_scope();
```

Comment:
Blocking: The following code should report an error for `o`. I think your code doesn't:

```javascript
with (o) {
}
let o = {};
```

Reply:
Yeah it didn't report error for the given code. I have moved `v.visit_enter_with_scope` below the errors.